android/expoview/src/main/java/host/exp/exponent/kernel/Kernel.java

```diff
@@ -493,6 +508,53 @@ public void handleIntent(Activity activity, Intent intent) {
       }
     }
 
+    openDefaultUrl();
+  }
+
+  private void openExperienceFromNotificationIntent(Intent intent) {
```

Comment:
Looking at this code: we both know it's not ideal. Do you think going for something like `boolean openExperienceFromNotificationIntent`, so we could write
```java
if (… && openExperienceFromNotificationIntent) {
  return;
}
...
```
would be worth it? It would kind of resemble iOS's `openURL:`. We wouldn't need to make this handler depend on `openDefaultUrl` being **the fallback to call**.